datacube_stats/main.py

```diff
@@ -446,11 +446,17 @@ def configure_outputs(self, index, metadata_type='eo') -> Dict[str, OutputProduc
         stats_metadata = _get_stats_metadata(self.config_file)
 
         for output_spec in self.output_product_specs:
+            definition = dict(output_spec)
+            if 'metadata' not in definition:
```

Comment:
What if `metadata` section exists, but is lacking `format.name` section.

I recommend using `toolz` to do that kind data tree surgery, probably `toolz.assoc_in` or `toolz.merge`

https://toolz.readthedocs.io/en/latest/api.html?highlight=assoc_in#dicttoolz